src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfo.ProjectData.cs

```diff
@@ -286,41 +288,37 @@ public static ProjectData Create(MSB.Execution.ProjectInstance projectInstance,
 
                 var projectReferences = ImmutableArray.CreateBuilder<string>();
                 var projectReferenceAliases = ImmutableDictionary.CreateBuilder<string, string>();
-                var projectReferencesAdded = new HashSet<string>();
-                foreach (var projectReferenceItem in projectInstance.GetItems(ItemNames.ProjectReference))
-                {
-                    var fullPath = projectReferenceItem.GetMetadataValue(MetadataNames.FullPath);
-
-                    if (IsCSharpProject(fullPath) && projectReferencesAdded.Add(fullPath))
-                    {
-                        projectReferences.Add(fullPath);
-
-                        var aliases = projectReferenceItem.GetMetadataValue(MetadataNames.Aliases);
-                        if (!string.IsNullOrEmpty(aliases))
-                        {
-                            projectReferenceAliases[fullPath] = aliases;
-                        }
-                    }
-                }
 
                 var references = ImmutableArray.CreateBuilder<string>();
                 var referenceAliases = ImmutableDictionary.CreateBuilder<string, string>();
                 foreach (var referencePathItem in projectInstance.GetItems(ItemNames.ReferencePath))
                 {
                     var referenceSourceTarget = referencePathItem.GetMetadataValue(MetadataNames.ReferenceSourceTarget);
+                    var aliases = referencePathItem.GetMetadataValue(MetadataNames.Aliases);
```

Comment:
I believe the old method `IsCSharpProject` can now be deleted

Reply:
Removed

Reply:
Actually it was probably necessary: if we have a reference coming from a VB project or a C++ project, we probably need to keep it as a metadata reference?

Reply:
Oh I see: there were two places this function checked if a project was a .csproj; I deleted the one using the function but the other one was never using it.

@filipw @JoeRobich any preference for having the .csproj check below call the function, versus just deleting it?

Reply:
@JoeRobich replied privately to keep it deleted.

Reply:
I'd say pragmatically that when usage count goes to 1, the function is no longer needed 🙂